code/datums/gamemode/factions/bloodcult/bloodcult.dm

```diff
@@ -0,0 +1,346 @@
+									"<span class='warning'>You hear a liquid flowing.</span>")
+		if ("grabbed")
+			var/mob/living/carbon/human/H = data["grabbed"]
+			blood = H.get_blood(H.vessel)
+			if (previous_result != "user")
+				user.visible_message("<span class='warning'>\The [user] stabs their nails inside \the [data["grabbed"]], drawing blood from them!</span>",
+									"<span class='rose'>You stab your nails inside \the [data["grabbed"]] to draw some blood from them.</span>",
+									"<span class='warning'>You hear a liquid flowing.</span>")
+		if ("bleeder")
+			var/mob/living/carbon/human/H = data["bleeder"]
+			blood = H.get_blood(H.vessel)
+			if (previous_result != "user")
+				user.visible_message("<span class='warning'>\The [user] dips their fingers inside \the [data["bleeder"]]'s wounds!</span>",
+									"<span class='rose'>You dip your fingers inside \the [data["bleeder"]]'s wounds to draw some blood from them.</span>",
+									"<span class='warning'>You hear a liquid flowing.</span>")
+		if ("held")
+			var/obj/item/weapon/reagent_containers/glass/G = data["held"]
+			blood = locate() in G.reagents.reagent_list
+			if (previous_result != "user")
+				user.visible_message("<span class='warning'>\The [user] tips \the [data["held"]], pouring blood!</span>",
+									"<span class='rose'>You tip \the [data["held"]] to pour the blood contained inside.</span>",
+									"<span class='warning'>You hear a liquid flowing.</span>")
+		if ("bloodpack")
+			var/obj/item/weapon/reagent_containers/blood/B = data["bloodpack"]
+			blood = locate() in B.reagents.reagent_list
+			if (previous_result != "user")
+				user.visible_message("<span class='warning'>\The [user] squeezes \the [data["bloodpack"]], pouring blood!</span>",
+									"<span class='rose'>You squeeze \the [data["bloodpack"]] to pour the blood contained inside.</span>",
+									"<span class='warning'>You hear a liquid flowing.</span>")
+		if ("container")
+			var/obj/item/weapon/reagent_containers/glass/G = data["container"]
+			blood = locate() in G.reagents.reagent_list
+			if (previous_result != "user")
+				user.visible_message("<span class='warning'>\The [user] dips their fingers inside \the [data["container"]], covering them in blood!</span>",
+									"<span class='rose'>You dip your fingers inside \the [data["container"]], covering them in blood.</span>",
+									"<span class='warning'>You hear a liquid flowing.</span>")
+		if ("user")
+			if (data["bloodpack_noholes"])
+				to_chat(user, "<span class='warning'>You must puncture \the [data["bloodpack"]] before you can squeeze blood from it!</span>")
+			else if (data["held_lid"])
+				to_chat(user, "<span class='warning'>Remove \the [data["held"]]'s lid first!</span>")
+			else if (data["container_lid"])
+				to_chat(user, "<span class='warning'>Remove \the [data["held"]]'s lid first!</span>")
+			var/mob/living/carbon/human/H = user
+			blood = H.get_blood(H.vessel)
```

Comment:
Well technically I'll have to make this an /atom proc because monkeys, constructs, and soul blades will all be able to store blood and make use of it.

Reply:
Actually I just checked and proc/get_blood() doesn't have anything to do with humans in particular, so I'm just turning it into a base proc.